docs/sdk-configuration.md

```diff
@@ -0,0 +1,676 @@
+    }
+
+    @Bean
+    @ForJaeger
+    public OpenTelemetry openTelemetryJaeger(TracerSdkProvider tracerProvider, MeterSdkProvider meterProvider) {
+        return OpenTelemetrySdk.builder()
+           .addPropagator(JaegerPropagator.getInstance())
+           .setTracerProvider(tracerProvider)
+           .setMeterProvider(meterProvider)
+           .build();
+    }
+  
+    @Bean
+    public AuthServiceStub authService(@ForJaeger OpenTelemetry openTelemetry, AuthConfig config) {
+        return AuthServiceGrpc.newBlockingStub(ManagedChannelBuilder.forEndpoint(config.getEndpoint()))
+          .withInterceptor(TracingClientInterceptor.create(openTelemetry));
+    }
+    
+    @Bean
+    public ServletFilter servletFilter(OpenTelemetry openTelemetry) {
+        return TracingServletFilter.create(openTelemetry);
+    }
+}
+
+// Use some instrumented client
+@Component
+public class MyAuthInterceptor {
+
+  private final AuthServiceStub authService;
+  
+  @Inject
+  public MyAuthInterceptor(AuthServiceStub authService) {
+    this.authService = authService;
+  }
+  
+  public void doAuth() {
+    if (!authService.getToken("credential").isAuthenticated()) {
+      throw new HackerException();
+    }
+  }
+}
+
+// Use tracer directly, not so common
+@Component
+public class MyService {
+
+  private final Tracer tracer;
+  private final Meter meter;
+  
+  @Inject
+  public MyService(TracerProvider tracerProvider, MeterProvider meterProvider) {
+    tracer = tracerProvider.get("my-service");
+    meter = meterProvider.get("my-service");
+  }
+  
+  public void doLogic() {
+    Span span = tracer.spanBuilder("logic").startSpan();
+    try (Scope ignored = span.makeCurrent()) {
+      Thread.sleep(1000);
+    } finally {
+      span.end();
+    }
+  }
+}
+```
+
+## The global instance of the SDK
+
+A built instance is convenient to use in most Java apps because of dependency injection. Because it
+has a easy-to-reason initialization ordering, being tied into the dependency ordering (even if dependency 
+injection happened to be done manually through constructor invocation), we encourage application
+developers to only use it.
+
+However, there are corner cases where an instance cannot be injected. The famous example is MySQL -
+MySQL interceptors are initialized by calling a default constructor, and there is no way to pass a
+built instance of a signal provider. For this case, we must store an SDK instance into a global
+variable. It is expected that frameworks or end-users will set the SDK as the global to support
+instrumentation that requires this.
+
+Before an SDK has been set, access to the global `OpenTelemetry` will return a no-op 
+`DefaultOpenTelemetry`. This is because it is possible library instrumentation is using the global,
+and it may even use it during the processing of a request (rather than only at initialization time).
+For this reason, we cannot throw an exception. Instead, if the SDK is detected on the classpath, we
```

Comment:
@jkwatson API-only case

Reply:
Thanks. This logging I think will be very helpful at finding where people are waiting until too late to set up their SDK. 👍